packages/@storefront/structure/src/filtered-list/index.ts

```diff
@@ -49,22 +53,39 @@ class FilteredList {
 
   updateItems(value: string = this.refs.filter.value) {
     value = value.trim().toLowerCase();
-    const filtered = this.props.items.filter((item) => {
-      if (!item) {
-        return false;
-      } else if (typeof item === 'string') {
-        return item.toLowerCase().includes(value);
-      } else if (typeof item.value === 'string') {
-        return item.value.toLowerCase().includes(value);
-      } else {
-        return false;
-      }
-    });
+
+    const filtered = this.props.items
+      .filter((item) => this.filterItem(value, item))
+      .map((item, _, items) => this.decorateItem(value, item, items));
 
     if (filtered.length !== 0 || this.state.items.length !== 0) {
       this.state.items = filtered;
     }
   }
+
+  filterItem(value: string, item: FilteredList.Item) {
+    if (!item) {
+      return false;
+    } else if (typeof item === 'string') {
+      return item.toLowerCase().includes(value);
+    } else if (typeof item.value === 'string') {
+      return item.value.toLowerCase().includes(value);
+    } else {
+      return false;
+    }
+  }
+
+  decorateItem(value: string, item: FilteredList.Item, items: FilteredList.Item[]) {
```

Comment:
This method doesn't quite feel right. I think it should take the index instead of the item itself, because right now there's no direct relation between `item` and `items` based on the signature.

Alternatively, don't accept `item` and do the `map` in here and return the array.

Reply:
What do you think about maintaining consistency between the initial parameters of `filterItem` and `decorateItem` (ie. `value`, and `item`), and updating the `decorateItem` signature to accept both `index` and `items` (ie. `(value, item, index, items)`? This way, the `decorateItem` signature becomes a superset of the array methods.

Reply:
If we were doing partial applications, I might agree with that, but even with that, it still feels off. That said, I'm not strongly opposed to keeping it in, but between the existing signature and your proposed signature, I'd go with the latter.

Reply:
Noted. I've implemented the latter approach (ie. `decorateItem`'s signature is now `(value, item, index, items)`).